File: app/src/main/java/com/leadinsource/dailyweightlog/db/DateConverter.java
package com.leadinsource.dailyweightlog.db;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

/**
 * Created by Matt on 23/02/2018.
 */

public class DateConverter {

    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }

}

File: app/src/main/java/com/leadinsource/dailyweightlog/db/DateConverter.kt
package com.leadinsource.dailyweightlog.db

import android.arch.persistence.room.TypeConverter

import java.util.Date

/**
 * Created by Matt on 23/02/2018.
 */

object DateConverter {

    @TypeConverter
    fun fromTimestamp(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? {
        return date?.time
    }

}
